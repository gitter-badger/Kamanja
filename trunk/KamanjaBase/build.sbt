name := "KamanjaBase"

version := "1.0"

scalaVersion := "2.10.4"

libraryDependencies += "log4j" % "log4j" % "1.2.17"

libraryDependencies += "org.json4s" %% "json4s-native" % "3.2.9" 

libraryDependencies += "org.json4s" %% "json4s-jackson" % "3.2.9" 

libraryDependencies += "com.google.guava" % "guava" % "16.0.1" 

libraryDependencies += "com.google.code.findbugs" % "jsr305" % "1.3.9"
