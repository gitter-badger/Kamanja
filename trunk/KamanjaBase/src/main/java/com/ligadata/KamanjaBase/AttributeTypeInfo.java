package com.ligadata.KamanjaBase;

public class AttributeTypeInfo {

	String name;
	int Index;
	TypeCategory typeCategory; // from TypeCategory
	byte valTypeId;
	byte keyTypeId;
	long valSchemaId;

	public AttributeTypeInfo(String name, int Index, TypeCategory typeCategory,
			byte valTypeId, byte keyTypeId, long valSchemaId) {
		this.name = name;
		this.Index = Index;
		this.typeCategory = typeCategory;
		this.valTypeId = valTypeId;
		this.valSchemaId = valSchemaId;
		this.keyTypeId = keyTypeId;
	}

	public enum TypeCategory {
		INT(0), STRING(1), FLOAT(2), DOUBLE(3), LONG(4), BYTE(5), CHAR(6), CONTAINER(
				1001), MAP(1002), ARRAY(1003);
		private int value;

		private TypeCategory(int value) {
			this.value = value;
		}

		public int getValue(){
			return this.value;
		}
	}

	public boolean IsContainer() {
		return (typeCategory == TypeCategory.CONTAINER);
	}

	public String getName() {
		return name;
	}

	public void setName(String name) {
		this.name = name;
	}

	public int getIndex() {
		return Index;
	}

	public void setIndex(int index) {
		Index = index;
	}

	public TypeCategory getTypeCategary() {
		return (TypeCategory) typeCategory;
	}

	public void setTypeCategary(TypeCategory typeCategory) {
		this.typeCategory = typeCategory;
	}

	public byte getValTypeId() {
		return valTypeId;
	}

	public void setValTypeId(byte valTypeId) {
		this.valTypeId = valTypeId;
	}

	public byte getKeyTypeId() {
		return keyTypeId;
	}

	public void setKeyTypeId(byte keyTypeId) {
		this.keyTypeId = keyTypeId;
	}

	public long getValSchemaId() {
		return valSchemaId;
	}

	public void setValSchemaId(long valSchemaId) {
		this.valSchemaId = valSchemaId;
	}

}
