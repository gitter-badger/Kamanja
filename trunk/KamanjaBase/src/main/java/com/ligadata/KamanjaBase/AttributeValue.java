package com.ligadata.KamanjaBase;

public class AttributeValue {

	String valueType = "";
	Object value = "";

	public String getValueType() {
		return this.valueType;
	}

	public void setValueType(String valType) {
		this.valueType = valType;
	}

	public Object getValue() {
		return this.value;
	}

	public void setValue(Object val) {
		this.value = val;
	}
}
