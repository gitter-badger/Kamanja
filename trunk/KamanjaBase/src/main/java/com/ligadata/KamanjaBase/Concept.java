package com.ligadata.KamanjaBase;

//FIXME:: Decide whether we need to generate concept class for every concept or use generic class like this
public class Concept implements ContainerOrConcept {
	String key = null;
	String valueType = null;
	Object value = null;

	Concept(String key, String valueType, Object value) {
		this.key = key;
		this.valueType = valueType;
		this.value = value;
	}
}