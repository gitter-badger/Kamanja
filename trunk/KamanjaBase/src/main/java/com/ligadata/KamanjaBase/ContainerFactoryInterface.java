package com.ligadata.KamanjaBase;

public abstract class ContainerFactoryInterface implements
		ContainerOrConceptFactory {

	enum ContainerType {
		MESSAGE, CONTAINER;
	}

	final public String getFullTypeName() {
		return (getTypeNameSpace() + "." + getTypeName());
	}

	final public boolean hasPrimaryKey() {
		String[] pKeys = getPrimaryKeyNames();
		return (pKeys != null && pKeys.length > 0);
	}

	final public boolean hasPartitionKey() {
		String[] pKeys = getPartitionKeyNames();
		return (pKeys != null && pKeys.length > 0);
	}

	final public boolean hasTimePartitionInfo() {
		TimePartitionInfo tmInfo = getTimePartitionInfo();
		return (tmInfo != null && tmInfo.timePartitionType != TimePartitionInfo.TimePartitionType.NONE);
	}

	public abstract String getTypeNameSpace();

	public abstract String getTypeName();

	public abstract String getTypeVersion();

	public abstract TimePartitionInfo getTimePartitionInfo();

	public abstract ContainerType getContainerType(); // ContainerType is enum of MESSAGE or CONTAINER

	public abstract boolean isFixed();

	public abstract String getSchema();

	public abstract String[] getPrimaryKeyNames();

	public abstract String[] getPartitionKeyNames();

	public abstract ContainerInterface createInstance();

}
