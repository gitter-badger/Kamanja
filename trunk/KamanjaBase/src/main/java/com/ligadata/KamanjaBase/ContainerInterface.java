package com.ligadata.KamanjaBase;

public abstract class ContainerInterface extends ContainerOrConcept {
	long transactionId = 0;
	long timePartitionData = 0;
	int rowNumber = 0;

	public ContainerInterface(ContainerFactoryInterface factory) {
		super(factory);
	}

	final ContainerFactoryInterface getContainerFactory() {
		return ((ContainerFactoryInterface) getFactory());
	}

	final public boolean hasPrimaryKey() {
		return getContainerFactory().hasPrimaryKey();
	}

	final public boolean hasPartitionKey() {
		return getContainerFactory().hasPartitionKey();
	}

	final public boolean hasTimePartitionInfo() {
		return getContainerFactory().hasTimePartitionInfo();
	}

	final public long getTransactionId() {
		return transactionId;
	}

	final public long getTimePartitionData() {
		return timePartitionData;
	}

	final public int getRowNumber() {
		return rowNumber;
	}

	final public int getSchemaId() {
		return getContainerFactory().getSchemaId();
	}

	final public ContainerFactoryInterface.ContainerType getContainerType() {
		return getContainerFactory().getContainerType();
	}

	final public boolean isFixed() {
		return getContainerFactory().isFixed();
	}

	final public String getAvroSchema() {
		return getContainerFactory().getAvroSchema();
	}

	final public String[] getPrimaryKeyNames() {
		return getContainerFactory().getPrimaryKeyNames();
	}

	final public String[] getPartitionKeyNames() {
		return getContainerFactory().getPartitionKeyNames();
	}

	final public TimePartitionInfo getTimePartitionInfo() {
		return getContainerFactory().getTimePartitionInfo();
	}

	final public void setTransactionId(long transId) {
		this.transactionId = transId;
	}

	final public void setTimePartitionData(long timePrtData) {
		this.timePartitionData = timePrtData;
	}

	final public void setRowNumber(int rowNbr) {
		this.rowNumber = rowNbr;
	}

	public abstract String[] getPrimaryKey();

	public abstract String[] getPartitionKey();

	public abstract void save();

	public abstract ContainerOrConcept Clone();

	public abstract String[] getAttributeNames();

	public abstract AttributeTypeInfo[] getAttributeTypes();

	public abstract Object get(String key); // Return (value, type)

	public abstract Object get(int index); // Return (value, type)

	public abstract Object getOrElse(String key, Object defaultVal); // Return
																		// (value,
																		// type)

	public abstract Object getOrElse(int index, Object defaultVal); // Return
																	// (value,
																	// type)

	public abstract AttributeValue[] getAllAttributeValues(); // Has (name,
																// value, type))

	public abstract void set(String key, Object value);

	public abstract void set(int index, Object value);

	public abstract void set(String key, Object value, String valTyp);

	public abstract java.util.Iterator<AttributeValue> getAttributeNameAndValueIterator();
}
