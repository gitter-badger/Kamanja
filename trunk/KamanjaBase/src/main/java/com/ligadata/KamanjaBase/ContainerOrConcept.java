package com.ligadata.KamanjaBase;

//FIXME:: Need to change ContainerOrConcept name to some meaningful name

public interface ContainerOrConcept {

}
