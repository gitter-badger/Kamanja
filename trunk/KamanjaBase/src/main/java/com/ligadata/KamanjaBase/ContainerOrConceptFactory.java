package com.ligadata.KamanjaBase;

public interface ContainerOrConceptFactory {

}
