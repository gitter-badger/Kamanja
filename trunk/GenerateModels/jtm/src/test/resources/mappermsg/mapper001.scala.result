/*
* Copyright 2016 ligaDATA
*
* Licensed under the Apache License, Version 2.0 (the "License");
* you may not use this file except in compliance with the License.
* You may obtain a copy of the License at
*
*     http://www.apache.org/licenses/LICENSE-2.0
*
* Unless required by applicable law or agreed to in writing, software
* distributed under the License is distributed on an "AS IS" BASIS,
* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
* See the License for the specific language governing permissions and
* limitations under the License.
*/
package com.ligadata.jtm.test.filter
import com.ligadata.KamanjaBase._
import com.ligadata.KvBase.TimeRange
import com.ligadata.kamanja.metadata.ModelDef
import com.ligadata.Utils._
class Factory(modelDef: ModelDef, nodeContext: NodeContext) extends ModelInstanceFactory(modelDef, nodeContext) {
  override def isValidMessage(msg: MessageContainerBase): Boolean = {
    msg.isInstanceOf[com.ligadata.kamanja.test.v1000000.varin1]
  }
  override def createModelInstance(): ModelInstance = return new Model(this)
  override def getModelName: String = "com.ligadata.jtm.test.filter"
  override def getVersion: String = "0.0.1"
  override def createResultObject(): ModelResultBase = new MappedModelResults()
}
class Model(factory: ModelInstanceFactory) extends ModelInstance(factory) {
  override def run(txnCtxt: TransactionContext, outputDefault: Boolean): Array[BaseMsg]  = {
    //
    //
    def exeGenerated_test1_1(msg1: com.ligadata.kamanja.test.v1000000.varin1): Array[BaseMsg] = {
      def process_o1(): Array[BaseMsg] = {
        val result = new com.ligadata.kamanja.test.v1000000.varout1
        result.timePartitionData = msg1.timePartitionData
        result.rowNumber = msg1.rowNumber
        result.out2 = msg1.in2
        result.out1 = msg1.in1
        result.transactionId = msg1.transactionId
        Array(result)
      }
      process_o1()
    }
    // Evaluate messages
    val msg1 = txnCtxt.getMessages("com.ligadata.kamanja.test.varin1").headOption.getOrElse(null).asInstanceOf[com.ligadata.kamanja.test.v1000000.varin1]
    // Main dependency -> execution check
    //
    val results: Array[BaseMsg] =
    if(msg1!=null) {
      exeGenerated_test1_1(msg1)
    } else {
      Array.empty[BaseMsg]
    }
    result
  }
}
