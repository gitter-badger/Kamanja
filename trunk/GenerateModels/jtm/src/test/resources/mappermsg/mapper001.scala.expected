/*
* Copyright 2016 ligaDATA
*
* Licensed under the Apache License, Version 2.0 (the "License");
* you may not use this file except in compliance with the License.
* You may obtain a copy of the License at
*
*     http://www.apache.org/licenses/LICENSE-2.0
*
* Unless required by applicable law or agreed to in writing, software
* distributed under the License is distributed on an "AS IS" BASIS,
* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
* See the License for the specific language governing permissions and
* limitations under the License.
*/
package com.ligadata.jtm.test.filter
import com.ligadata.KamanjaBase._
import com.ligadata.KvBase.TimeRange
import com.ligadata.kamanja.metadata.ModelDef
import com.ligadata.Utils._
class Factory(modelDef: ModelDef, nodeContext: NodeContext) extends ModelInstanceFactory(modelDef, nodeContext) {
  override def isValidMessage(msg: MessageContainerBase): Boolean = {
    msg.isInstanceOf[com.ligadata.kamanja.test.v1000000.varin1]
  }
  override def createModelInstance(): ModelInstance = return new Model(this)
  override def getModelName: String = "com.ligadata.jtm.test.filter"
  override def getVersion: String = "0.0.1"
  override def createResultObject(): ModelResultBase = new MappedModelResults()
}
class Model(factory: ModelInstanceFactory) extends ModelInstance(factory) {
  override def run(txnCtxt: TransactionContext, outputDefault: Boolean): Array[BaseMsg]  = {
    //
    //
    def exeGenerated_test1_1(msg1: com.ligadata.kamanja.test.v1000000.varin1): Array[BaseMsg] = {
      // in scala, type could be optional
      val out3: Int = msg1.get("in3") + 1000
      def process_o1(): Array[BaseMsg] = {
        if (!(out3 != -1 && out3 < 100)) return Array.empty[BaseMsg]
        val t1: String = "s:" + msg1.get("in3").toString()
        val result = new com.ligadata.kamanja.test.v1000000.varout1
        result.out2 = msg1.in2
        result.rowNumber = msg1.rowNumber
        result.timePartitionData = msg1.timePartitionData
        result.out1 = msg1.in1
        result.transactionId = msg1.transactionId
        result.set("out3", t1)
        result.set("out4", msg1.get("in4"))
        Array(result)
      }
      process_o1()
    }
    // Evaluate messages
    val msg1 = txnCtxt.getMessages("com.ligadata.kamanja.test.varin1").headOption.getOrElse(null).asInstanceOf[com.ligadata.kamanja.test.v1000000.varin1]
    // Main dependency -> execution check
    //
    val results: Array[BaseMsg] =
    if(msg1!=null) {
      exeGenerated_test1_1(msg1)
    } else {
      Array.empty[BaseMsg]
    }
    result
  }
}
